roofit/histfactory/src/ConfigParser.cxx

```diff
@@ -274,115 +272,98 @@ HistFactory::Measurement ConfigParser::CreateMeasurementFromDriverNode( TXMLNode
   measurement.SetBinHigh( 1 );
   measurement.SetExportOnly( false );
 
-  cxcoutIHF << "Creating new measurement: " << std::endl;
+  cxcoutIHF << "Creating new measurement: \n";
 
   // First, get the attributes of the node
   TListIter attribIt = node->GetAttributes();
-  TXMLAttr* curAttr = 0;
-  while( ( curAttr = dynamic_cast< TXMLAttr* >( attribIt() ) ) != 0 ) {
-
-    if( curAttr->GetName() == TString( "" ) ) {
-      cxcoutEHF << "Found XML attribute in Measurement with no name"  << std::endl;
+  TXMLAttr* curAttr = nullptr;
+  while ( (/**/curAttr = dynamic_cast<TXMLAttr*>(attribIt())/**/) != nullptr ) {
+    const char * curAttrName = curAttr->GetName();  // curAttr is guaranteed non-null
+    const char * curAttrValue = curAttr->GetValue();
```

Comment:
How about making those `const std::string` and ...

Reply:
good idea